pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -372,6 +374,15 @@ func (c *InstallerController) manageInstallationPods(ctx context.Context, operat
 
 		// if we are in a transition, check to see whether our installer pod completed
 		if currNodeState.TargetRevision > currNodeState.CurrentRevision {
+			if currNodeState.LastFailedRevision == currNodeState.TargetRevision && currNodeState.LastFailedTime != nil && !currNodeState.LastFailedTime.IsZero() {
+				delay := c.installerBackOff(currNodeState.LastFailedCount)
+				earliestRetry := currNodeState.LastFailedTime.Add(delay)
+				if !c.now().After(earliestRetry) {
+					klog.V(4).Infof("Backing off node %s installer retry %d until %v", currNodeState.NodeName, currNodeState.LastFailedCount+1, earliestRetry)
+					return true, earliestRetry.Sub(c.now()), nil
```

Comment:
so:
1. we obtain the delay based on the backoff function and the  `currNodeState.LastFailedCount`
2. we don't retry until delay sec after `currNodeState.LastFailedTime`

my questions is what happens if current clock and `currNodeState.LastFailedTime`  are not in sync, i.e. edge case, controller restarts and node clock goes crazy and goes back in time or DST change, , ...

Reply:
we don't support that case. Clocks not in sync break OpenShift and Kubernetes.